dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeUtils.cs

```diff
@@ -0,0 +1,261 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SemanticKernel.Connectors.Memory.Pinecone.Http.ApiSchema;
+using Microsoft.SemanticKernel.Connectors.Memory.Pinecone.Model;
+
+namespace Microsoft.SemanticKernel.Connectors.Memory.Pinecone;
+
+public static class PineconeUtils
+{
+    /// <summary>
+    ///  The maximum size of the metadata associated with each vector.
+    ///  See https://docs.pinecone.io/docs/metadata-filtering#supported-metadata-size
+    /// </summary>
+    public const int MaxMetadataSize = 40 * 1024;
+
+    /// <summary>
+    ///  The default dimension for Pinecone vectors. Equivalent to text-embeddings-ada-002 dimension.
+    /// </summary>
+    public const int DefaultDimension = 1536;
```

Comment:
I'd prefer to have this as a required constructor to avoid affinity to ADA

Reply:
MaxMetaDataSize is a constraint set by Pinecone.

DefaultDimension is only used in the creation of a Default IndexDefinition:

```
  public static IndexDefinition Default(string? name = default)
    {
        string indexName = name ?? PineconeUtils.DefaultIndexName;

        return Create(indexName)
            .WithDimension(PineconeUtils.DefaultDimension)
            .WithMetric(PineconeUtils.DefaultIndexMetric)
            .NumberOfPods(1)
            .NumberOfReplicas(1)
            .WithPodType(PineconeUtils.DefaultPodType)
            .WithMetadataIndex(MetadataIndexConfig.Default);
    }
```

I can move all Default IndexDefinition values to the IndexDefinition class file if you prefer.

Reply:
DefaultDimension is only used in the creation of a Default IndexDefinition:

```
  public static IndexDefinition Default(string? name = default)
    {
        string indexName = name ?? PineconeUtils.DefaultIndexName;

        return Create(indexName)
            .WithDimension(PineconeUtils.DefaultDimension)
            .WithMetric(PineconeUtils.DefaultIndexMetric)
            .NumberOfPods(1)
            .NumberOfReplicas(1)
            .WithPodType(PineconeUtils.DefaultPodType)
            .WithMetadataIndex(MetadataIndexConfig.Default);
    }
```

I can move all Default IndexDefinition values to the IndexDefinition class file if you prefer.